src/Microsoft.SqlTools.Hosting/Utility/LongList.cs

```diff
@@ -138,6 +143,34 @@ public T GetItem(long index)
         }
 
         /// <summary>
+        /// Sets the item at the specified index
+        /// </summary>
+        /// <param name="index">Index of the item to set</param>
+        /// <param name="value">The item to store at the index specified</param>
+        public void SetItem(long index, T value)
+        {
+            if (Count <= this.ExpandListSize)
+            {
+                int i32Index = Convert.ToInt32(index);
+                shortList[i32Index] = value;
+            }
+            else
+            {
+                int iArray32Index = (int) (Count / this.ExpandListSize);
+                if (expandedList.Count > iArray32Index)
```

Comment:
Hmm... This was a slight modification of the code I brought over from the SSMS code. I mostly just trusted they knew what they were doing, but now that you mention it, this looks a little odd. I'll give it a once over.